src/Models/ProductModel.php

```diff
@@ -0,0 +1,11 @@
+<?php
+class ProductModel {
+public static function getProducts(PDO $db, int $id) : array
+{
+    $sql = 'SELECT `price`, `color`, `stock` FROM `products` WHERE `category_id` = :id;';
+    $query = $db->prepare($sql);
+    $query->setFetchMode(PDO::FETCH_CLASS, ProductsEntity::class);
+    $query->execute(['id'=>$id]);
+    return $query->fetchAll();
+}
```

Comment:
Naming does not make it very clear what this function does or what you need to put into it.

```suggestion
public static function getProductsByCategoryId(PDO $db, int $category_id) : array
{
    $sql = 'SELECT `price`, `color`, `stock` FROM `products` WHERE `category_id` = :category_id;';
    $query = $db->prepare($sql);
    $query->setFetchMode(PDO::FETCH_CLASS, ProductsEntity::class);
    $query->execute(['category_id'=>$category_id]);
    return $query->fetchAll();
}
```